src/components/RegisterPage/RegisterPage.tsx

```diff
@@ -0,0 +1,470 @@
+	const { t } = useTranslation('LoginRegisterPage');
+	const appDispatch = useAppDispatch();
+	const [toolboxEnabled, setToolboxEnabled] = useState<boolean>(false);
+	const [passwordShow, setPasswordShow] = useState(false);
+	const [passwordConfirmShow, setPasswordConfirmShow] = useState(false);
+	const [passwordIcon, setPasswordIcon] = useState(eye);
+	const [passwordConfirmIcon, setPasswordConfirmIcon] = useState(eye);
+	const validationSchema = Yup.object({
+		firstName: validateFirstName(t),
+		lastName: validateLastName(t),
+		email: validateEmail(t),
+		phoneNumber: validatePhoneNumber(t),
+		password: validatePassword(t),
+		passwordConfirm: validatePasswordConfirm(t),
+		readTerms: validateReadTerms(t),
+		...(toolboxEnabled
+			? {
+					companyName: validateCompanyName(t),
+					position: validatePosition(t),
+					taxNumber: validateTaxNumber(t),
+					postIndex: validatePostIndex(t),
+					city: validateCity(t),
+					street: validateStreet(t),
+					houseNumber: validateHouseNumber(t),
+			  }
+			: {}),
+	});
+
+	const handleRegister = (formValues: {
+		firstName: string;
+		lastName: string;
+		email: string;
+		phoneNumber: string;
+		password: string;
+		companyName: string;
+		position: string;
+		taxNumber: string;
+		postIndex: string;
+		city: string;
+		street: string;
+		houseNumber: string;
+	}): void => {
+		appDispatch(register(formValues))
+			.then((res) => {
+				const payload = res.payload as { message: string };
+				if (payload.message) {
+					// Дальнейшие действия после успешной регистрации
+				}
+			})
+			.catch(() => {
+				// Обработка ошибки при регистрации
+			});
+	};
+	const handlePasswordToggle: MouseEventHandler<HTMLDivElement> = (event) => {
+		event.preventDefault();
+		setPasswordShow(!passwordShow);
+		setPasswordIcon(passwordShow ? eye : eyeBlocked);
+	};
+
+	const handlePasswordConfirmToggle: MouseEventHandler<HTMLDivElement> = (event) => {
+		event.preventDefault();
+		setPasswordConfirmShow(!passwordConfirmShow);
+		setPasswordConfirmIcon(passwordConfirmShow ? eye : eyeBlocked);
+	};
+
+	const USER_LABEL = (
+		<div className="mt-4">
+			<div className="login__account--divide position__relative d-flex justify-content-center">
+				<span className="login__account--divide__text text-black">{t('user_line')}</span>
+			</div>
+		</div>
+	);
+	const firstName = (
+		<Form.Group as={Col} id="formUsername">
+			{USER_LABEL}
+			<InputGroup>
+				<Field as={FloatingInput} id="firstName" name="firstName" placeholder={t('username')} />
+			</InputGroup>
+			<ErrorMessage name="firstName" component="div" className="validation_warning__message" />
+		</Form.Group>
+	);
+	const lastName = (
+		<Form.Group as={Col} id="formLastName">
+			<InputGroup>
+				<Field as={FloatingInput} id="lastName" name="lastName" placeholder={t('last_name')} />
+			</InputGroup>
+			<ErrorMessage name="lastName" component="div" className="validation_warning__message" />
+		</Form.Group>
+	);
```

Comment:
create componet for all  firstName, lastName add other

```
 newComponent (props) =>
 const {label, asFiels, id, name, placeholder, component, className} = props
```

Reply:
done